mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/mapping/building/impl/PojoMapper.java

```diff
@@ -93,19 +96,21 @@ public PojoMapper(MappingBuildContext buildContext,
 			TypeMetadataContributorProvider<PojoTypeMetadataContributor> contributorProvider,
 			PojoBootstrapIntrospector introspector,
 			ContainerExtractorRegistry containerExtractorRegistry,
-			boolean implicitProvidedId,
+			BeanReference<? extends IdentifierBridge<Object>> identifierBridge,
 			boolean multiTenancyEnabled,
 			PojoMapperDelegate<MPBS> delegate) {
 		this.failureCollector = buildContext.getFailureCollector();
 		this.contributorProvider = contributorProvider;
-		this.implicitProvidedId = implicitProvidedId;
 		this.multiTenancyEnabled = multiTenancyEnabled;
 
 		this.failureHandler = buildContext.getFailureHandler();
 		this.threadPoolProvider = buildContext.getThreadPoolProvider();
 
 		this.delegate = delegate;
 
+		this.identifierBridge = ( identifierBridge == null ) ? null :
+				buildContext.getBeanResolver().resolve( identifierBridge );
```

Comment:
A `BeanHolder` must be closed. If you instantiate the bridge here, you must close it in case of failure.

Personally, I would just store the reference here, pass it along to `new PojoIndexedTypeManagerBuilder` in `createIndexedTypeManagerBuilder`, and only instantiate the bridge in `PojoIdentityMappingCollectorImpl#applyDefaults` *if necessary*. You really don't need to instantiate the bridge so early.

This is actually what was explained in the ticket, so I assume you did this because you didn't have access to the bean resolver in `PojoIdentityMappingCollectorImpl`. If that's the problem, well, I don't have any problem with passing the bean resolver or the `MappingBuildContext` alongside the bean reference.

It's also more correct, given that the APIs allow users to override the provided ID on a per-entity basis, and the map the ID from a property instead. Though that's not really a feature, more a side-effect of how the APIs are designed.